apps/cowswap-frontend/src/modules/tokensList/pure/TokenTags/index.tsx

```diff
@@ -12,51 +13,58 @@ interface TagInfo {
   name: string
   description: string
   icon?: string
+  color?: StatusColorVariant
 }
 
-enum Tags {
-  UNSUPPORTED = '0',
-  GAS_FREE = '1',
-}
+export type TokenTagType = keyof typeof TOKEN_TAGS
 
-const TOKEN_TAGS: Record<Tags, TagInfo> = {
-  [Tags.UNSUPPORTED]: {
+const TOKEN_TAGS: Record<string, TagInfo> = {
+  unsupported: {
     name: 'Unsupported',
     description:
       'This token is unsupported as it does not operate optimally with CoW Protocol. Please refer to the FAQ for more information.',
     id: '0',
+    color: StatusColorVariant.Warning,
   },
-  [Tags.GAS_FREE]: {
-    name: 'Gas-free approval',
+  'gas-free': {
+    name: 'Gas-free',
     icon: ICON_GAS_FREE,
     description: 'This token supports gas-free approvals. Enjoy! 🐮',
     id: '1',
+    color: StatusColorVariant.Success,
+  },
+  circle: {
+    name: 'Circle Native',
+    description: 'Token officially issued by Circle',
+    id: '2',
+    color: StatusColorVariant.Info,
   },
 }
 
 export function TokenTags({
   isUnsupported,
   isPermitCompatible,
+  tags = [],
 }: {
   isUnsupported: boolean
   isPermitCompatible?: boolean
+  tags?: string[]
 }) {
-  const tagsToShow: TagInfo[] = []
-
-  if (isUnsupported) {
-    tagsToShow.push(TOKEN_TAGS[Tags.UNSUPPORTED])
-  } else if (isPermitCompatible) {
-    tagsToShow.push(TOKEN_TAGS[Tags.GAS_FREE])
-  }
+  const tagsToShow = [
+    // Include valid tags from token.tags
+    ...tags.filter((tag) => tag in TOKEN_TAGS).map((tag) => TOKEN_TAGS[tag as TokenTagType]),
+    // Add unsupported tag if applicable
+    ...(isUnsupported ? [TOKEN_TAGS.unsupported] : []),
+    // Add gas-free tag if applicable (and not unsupported)
+    ...(!isUnsupported && isPermitCompatible ? [TOKEN_TAGS['gas-free']] : []),
+  ]
```

Comment:
I think `unsupported` should take precedence.
If it's set, don't show any other tag.
Maybe something like:

```suggestion
  const tagsToShow = isUnsupported ? [TOKEN_TAGS.unsupported] : [
    // Include valid tags from token.tags
    ...tags.filter((tag) => tag in TOKEN_TAGS).map((tag) => TOKEN_TAGS[tag as TokenTagType]),
    // Add gas-free tag if applicable
    ...(isPermitCompatible ? [TOKEN_TAGS['gas-free']] : []),
  ]
```

Reply:
Yes makes sense. If a token is unsupported other will either not apply anyway (e.g. Circle) or not relevant (Gas-free). Implemented now.